plugin-locationlayer/src/main/java/com/mapbox/mapboxsdk/plugins/locationlayer/LocationLayer.java

```diff
@@ -192,15 +192,22 @@ private void setLayerVisibility(String layerId, boolean visible) {
     }
   }
 
-  private void addLayers() {
-    addSymbolLayer(SHADOW_LAYER, BACKGROUND_LAYER);
-    addSymbolLayer(BACKGROUND_LAYER, FOREGROUND_LAYER);
-    addSymbolLayer(FOREGROUND_LAYER, null);
-    addSymbolLayer(BEARING_LAYER, null);
+  private void addLayers(LocationLayerOptions options) {
+    addSymbolLayer(SHADOW_LAYER, BACKGROUND_LAYER, options);
+    addSymbolLayer(BACKGROUND_LAYER, FOREGROUND_LAYER, options);
+    addSymbolLayer(FOREGROUND_LAYER, null, options);
+    addSymbolLayer(BEARING_LAYER, null, options);
     addAccuracyLayer();
   }
 
-  private void addSymbolLayer(String layerId, String beforeLayerId) {
+  private void addSymbolLayer(String layerId, String beforeLayerId, LocationLayerOptions options) {
+    String foregroundString = options.gpsName() != null ? options.gpsName() : options.foregroundName();
```

Comment:
@LukasPaczos ran into a possible issue here - the GPS / Foreground drawables both share the foreground icon layer.  At the time we add the symbol layer here, we don't know what render mode we are in yet.  So a decision needs to be made as far as what String is used.  Here I prioritized the GPS name.  What do you think?

Reply:
I think the resolution here would be keeping the string IDs of the used images in the `locationFeature`, this way we would be able to update them in the runtime, without reloading the map.